src/gui/inputdevices/StylusInputHandler.cpp

```diff
@@ -92,8 +90,10 @@ auto StylusInputHandler::handleImpl(InputEvent const& event) -> bool {
     // Trigger end of action if pen tip leaves screen or mouse button is released
     if (event.type == BUTTON_RELEASE_EVENT) {
         if (event.button == 1 || this->inputContext->getSettings()->getInputSystemTPCButtonEnabled()) {
-            if (this->inputRunning) {
+            if (this->eventsToIgnore < 0) {
                 this->actionEnd(event);
+            } else {
+                this->eventsToIgnore = -1;
```

Comment:
Should this not be done in either case?

Reply:
It is -1 already in the first case. But i can also set it every time if you want?

Reply:
Nah if it is -1 already then its good.